core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -214,7 +214,10 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         }
 
         // check if new domain is wider than domain already provided by table scan
-        if (constraint.predicate().isEmpty() && newDomain.contains(node.getEnforcedConstraint())) {
+        if (constraint.predicate().isEmpty() &&
+                // TODO do we need to track enforced ConnectorExpression in TableScanNode?
+                TRUE.equals(connectorExpression.orElse(TRUE)) &&
```

Comment:
IMO this is clearer:
`Optional.of(TRUE).equals(connectorExpression)`

Reply:
`connectorExpression.orElse(TRUE)` is already used in oither places in this method.